### PR TITLE
Introduce a random battles mode

### PR DESCRIPTION
**Pull request by AustinXII**

### CODE REVIEW

File: src/honkalculate.template.html [hunk 1]
```diff
@@ -57,7 +57,8 @@
         <span title="Select the calculator's mode of function.">
             <input class="mode visually-hidden" type="radio" name="mode" id="all-vs-one" /><label class="btn btn-wide btn-right right" for="all-vs-one">All vs One</label>
             <input class="mode visually-hidden" type="radio" name="mode" id="one-vs-all" /><label class="btn btn-wide btn-mid right" for="one-vs-all">One vs All</label>
-            <input class="mode visually-hidden" type="radio" name="mode" id="one-vs-one" /><label class="btn btn-wide btn-left right" for="one-vs-one">One vs One</label>
+            <input class="mode visually-hidden" type="radio" name="mode" id="one-vs-one" /><label class="btn btn-wide btn-mid right" for="one-vs-one">One vs One</label>
+            <input class="mode visually-hidden" type="radio" name="mode" id="randoms" /><label class="btn btn-wide btn-left right" for="randoms">Random Battles</label>
```
Comment (scheibo) [inline src/honkalculate.template.html]: Please sure this toggle doesn't distort at different browser sizes (you'll probably need to tweak the CSS - i think it already distorts on occasion even without the new toggle)

File: src/js/shared_controls.js [hunk 2]
```diff
@@ -534,7 +537,8 @@ function createPokemon(pokeInfo) {
 	if (typeof pokeInfo === "string") { // in this case, pokeInfo is the id of an individual setOptions value whose moveset's tier matches the selected tier(s)
 		var name = pokeInfo.substring(0, pokeInfo.indexOf(" ("));
 		var setName = pokeInfo.substring(pokeInfo.indexOf("(") + 1, pokeInfo.lastIndexOf(")"));
-		var set = setdex[name][setName];
+		var isRandoms = mode === 'randoms';
+		var set = isRandoms ? randdex[name][setName] : setdex[name][setName];
```
Comment (scheibo) [inline src/js/shared_controls.js]: `var set = (isRandoms ? randdex : setdex)[name][setName];` also works, not really preference though

Comment (scheibo) [reply src/js/shared_controls.js]: Also, I dont think  sets have names? the random sets JSON is different than the normal sets.

File: src/js/shared_controls.js [hunk 3]
```diff
@@ -863,6 +863,9 @@ function getSetOptions(sets) {
 					text: pokeName + " (" + "Randoms" + ")",
 					id: pokeName + " (" + "Randoms" + ")"
 				});
+				if (!pokeName in randdex) {
```
Comment (scheibo) [inline src/js/shared_controls.js]: rm

Comment (AustinXII) [reply src/js/shared_controls.js]: ?

**Pull request merged**